packages/core/src/entrypoint/types.ts

```diff
@@ -1,9 +1,20 @@
 import type { Address, Chain, Hex } from "viem";
+import type { UserOperationRequest } from "../types";
 
-export type EntryPointDef<EntryPointUserOperation> = {
-  version: string;
+export type EntryPointDef<
+  TEntryPointVersion extends EntryPointVersion | undefined =
+    | EntryPointVersion
+    | undefined
+> = {
+  version: TEntryPointVersion;
```

Comment:
I don't think we need to make this generic tbh... this version field in the def can just be `EntrypointVersion` and then impls can be more restrictive in their typing by doing something like `version: 0.6.0 as const` or `version: 0.7.0 as const`